src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -81,12 +85,43 @@ public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
+    public Answers getAnswers() {
         return answers;
     }
 
     @Override
     public String toString() {
         return "Question [id=" + getId() + ", title=" + title + ", contents=" + contents + ", writer=" + writer + "]";
     }
+
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        validateDelete(loginUser);
+        this.deleted = true;
+
+        List<DeleteHistory> deleteHistories = new ArrayList<>(Collections.singletonList(makeDeleteHistory()));
+        deleteHistories.addAll(deleteAnswers(loginUser));
+        return deleteHistories;
+    }
+
+    private void validateDelete(NsUser loginUser) throws CannotDeleteException {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("로그인 사용자와 질문한 사용자가 같아야 합니다.");
+        }
+    }
+
+    private List<DeleteHistory> deleteAnswers(NsUser loginUser) throws CannotDeleteException {
+        if (Objects.nonNull(this.answers)) {
```

Comment:
```suggestion
        if (answers != null) {
```
null 체크를 동등연산자를 사용해서 체크할 수 있었을텐데
Objects의 `nonNull` 메소드를 사용하셨네요

<img width="636" alt="image" src="https://github.com/next-step/java-lms/assets/36617783/40e4e74c-d291-48f0-8a82-7f3f1265379a">

문서를 보시면 `This method exists to be used as a java.util.function.Predicate, filter(Objects::nonNull)`
람다 표현식을 간단하게 표시할 때 사용하라고 만든 메서드입니다.
null을 직접 체크하는 방법도 의미전달이 충분히 되기 때문에 굳이 메소드 콜을 한번더 할 필요는 없을 것 같다는 생각이 듭니다.